tests/db_engine_specs_test.py

```diff
@@ -80,3 +81,45 @@ def test_job_2_launched_stage_2_stages_progress(self):
             17/02/07 19:16:09 INFO exec.Task: 2017-02-07 19:16:09,173 Stage-1 map = 40%,  reduce = 0%
         """.split('\n')  # noqa ignore: E501
         self.assertEquals(60, HiveEngineSpec.progress(log))
+
+    def test_wrapped_query(self):
+        sql = 'SELECT * FROM a'
+        db = Database(sqlalchemy_uri='mysql://localhost')
+        limited = MssqlEngineSpec.apply_limit_to_sql(sql, 1000, db)
+        expected = 'SELECT * \nFROM (SELECT * FROM a) AS inner_qry \n LIMIT 1000'
+        self.assertEquals(expected, limited)
+
+    def test_simple_limit_query(self):
+        sql = 'SELECT * FROM a'
+        db = Database(sqlalchemy_uri='mysql://localhost')
+        limited = MySQLEngineSpec.apply_limit_to_sql(sql, 1000, db)
+        expected = 'SELECT * FROM a LIMIT 1000'
+        self.assertEquals(expected, limited)
+
+    def test_modify_limit_query(self):
+        sql = 'SELECT * FROM a LIMIT 9999'
+        db = Database(sqlalchemy_uri='mysql://localhost')
+        limited = MySQLEngineSpec.apply_limit_to_sql(sql, 1000, db)
+        expected = 'SELECT * FROM a LIMIT 1000'
+        self.assertEquals(expected, limited)
+
+    def test_modify_newline_query(self):
+        sql = 'SELECT * FROM a\nLIMIT 9999'
+        db = Database(sqlalchemy_uri='mysql://localhost')
+        limited = MySQLEngineSpec.apply_limit_to_sql(sql, 1000, db)
+        expected = 'SELECT * FROM a LIMIT 1000'
+        self.assertEquals(expected, limited)
+
+    def test_modify_lcase_limit_query(self):
+        sql = 'SELECT * FROM a\tlimit 9999'
+        db = Database(sqlalchemy_uri='mysql://localhost')
+        limited = MySQLEngineSpec.apply_limit_to_sql(sql, 1000, db)
+        expected = 'SELECT * FROM a LIMIT 1000'
+        self.assertEquals(expected, limited)
+
+    def test_limit_query_with_limit_subquery(self):
+        sql = 'SELECT * FROM (SELECT * FROM a LIMIT 10) LIMIT 9999'
+        db = Database(sqlalchemy_uri='mysql://localhost')
+        limited = MySQLEngineSpec.apply_limit_to_sql(sql, 1000, db)
+        expected = 'SELECT * FROM (SELECT * FROM a LIMIT 10) LIMIT 1000'
+        self.assertEquals(expected, limited)
```

Comment:
I'm nervous about the regular expression, I'd add a few more unit tests covering some weird cases, eg:

```sql
SELECT
    'LIMIT 1000' AS a
  , b
FROM
    table
LIMIT
    1000
<EOF>
```

And maybe some stuff with extra space at the end:

```sql
...
LIMIT      1000  ;    
  <EOF>
```